pkg/cloudprovider/provider/gce/config.go

```diff
@@ -251,16 +262,28 @@ func (cfg *config) postprocessServiceAccount() error {
 		sa = string(decoded)
 	}
 
+	creds, err := googleoauth.CredentialsFromJSON(context.TODO(), []byte(sa), compute.ComputeScope)
```

Comment:
Nit: it might be better to use `context.Background()` here.

Reply:
Why? This transaction is actually propagated by another caller which is the cloud providers methods. `context.Background()` which is used at main functionality(e.g: machine creating), however, `context.TODO()` is used when the main context isn't propagated to the called functions ;)

Reply:
`context.TODO()` is a kinda "marker" to get back to this and put a real context (e.g. pass context down the stack or use `context.Background()`). If we don't plan on refactoring this later to provide already existing context, using `context.Background()` is a semantically better approach. However, I'm not going to block on this, I don't have strong opinions.

Reply:
Well that's the thing, I wanted to come back again and propagate the context through all other internal functions that needed it ;)